src/neo/Ledger/ContractState.cs

```diff
@@ -74,6 +90,8 @@ public static ContractState FromJson(JObject json)
             ContractState contractState = new ContractState();
             contractState.Script = Convert.FromBase64String(json["script"].AsString());
             contractState.Manifest = ContractManifest.FromJson(json["manifest"]);
+            contractState.RedirectionHash = UInt160.Parse(json["redirectionHash"].AsString());
+            contractState.HasUpgraded = json["hasUpgraded"].AsBoolean();
```

Comment:
I agree with @shargon, perhaps `WasUpgraded` will be better.

Reply:
Or just `Upgraded`

Reply:
OK,I will adjust